pkg/apis/core/validation/cloudprofile.go

```diff
@@ -44,11 +45,50 @@ func ValidateCloudProfileUpdate(newProfile, oldProfile *core.CloudProfile) field
 	allErrs := field.ErrorList{}
 
 	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newProfile.ObjectMeta, &oldProfile.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateCloudProfileSpecUpdate(&newProfile.Spec, &oldProfile.Spec, field.NewPath("spec"))...)
 	allErrs = append(allErrs, ValidateCloudProfile(newProfile)...)
 
 	return allErrs
 }
 
+// ValidateCloudProfileSpecUpdate validates the spec update of a CloudProfile
+func ValidateCloudProfileSpecUpdate(new, old *core.CloudProfileSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateCloudProfileVersionsUpdate(new.Kubernetes.Versions, old.Kubernetes.Versions, fldPath.Child("kubernetes", "versions"))...)
+
+	for _, oldImage := range old.MachineImages {
+		for index, newImage := range new.MachineImages {
+			if oldImage.Name == newImage.Name {
+				allErrs = append(allErrs, validateCloudProfileVersionsUpdate(newImage.Versions, oldImage.Versions, fldPath.Child("machineImages").Index(index).Child("versions"))...)
+			}
+		}
+	}
+
+	return allErrs
+}
+
+// ValidateCloudProfileAddedVersions validates versions added to the CloudProfile
+func ValidateCloudProfileAddedVersions(versions []core.ExpirableVersion, addedVersions map[string]int, fldPath *field.Path) field.ErrorList {
```

Comment:
To my understanding, If this check is  in the Spec validation, than that would block any  update of the CloudProfile when containing an expired version. I do not think forcing the CloudProfile to be expired version free is operator friendly. 
Rather I would only validate that when adding new version to an existing profile. 

I think that the underlying problem is that,  when updating the CloudProfile the [same checks as when initially creating the CloudProfile are also performed](https://github.com/gardener/gardener/blob/90b9eb1905851dadd3713d2e65bfbe7b5717a03f/pkg/apis/core/validation/cloudprofile.go#L49).

Reply:
> I do not think forcing the CloudProfile to be expired version free is operator friendly.

I don't understand why it's operator friendly to allow specifying expired version during creation whereas it's not for updates (or the other way around). What's the use case to add an already expired version through creation and what's the motivation to prevent it thru updates?

Reply:
> I think that the underlying problem is that, when updating the CloudProfile the same checks as when initially creating the CloudProfile are also performed.
The problem is, the way the validation **is designed at the moment:**
- Create: ValidateCloudProfileSpec
- Update: ValidateCloudProfileSpecUpdate() & ValidateCloudProfileSpec()

So if I want to introduce a check during CloudProfile Create() I have to check that in the ValidateCloudProfileSpec(). As newly created version do not have previous field values,  I would have to introduce a check that no version can have an expiration date in the past. 

No comes the problem: A version X has an expiration date Y. Time passes after Y and the operator wants to make a change to the CloudProfile completely unrelated to the versions. That will not work as the validation will throw an error that the CloudProfile is not valid. The operator has to fix the expired version first before being able to perform his intendet CloudProfile update. Thats what I meant with "not operator friendly".

To sum it up: The way the validation is currently designed is in my opinion causing problems here. There is a coupling between the update validation logic of the CloudProfile and the Creation validation logic.

I propose changing the validation logic by introducing a separate function ValidateCloudProfileCreation() [that calls the current validation function  ](https://github.com/gardener/gardener/blob/90b9eb1905851dadd3713d2e65bfbe7b5717a03f/pkg/apis/core/validation/cloudprofile.go#L34).
This way we can validate in the ValidateCloudProfileCreation() that no version has an expiration date in the past and in the Update we do the same for added versions.
I ll update the PR accordingly.

Reply:
Sounds good 👍

Reply:
I updated it - [PTAL here](https://github.com/gardener/gardener/blob/0921db6fe993a98c2eff652588ded6fedabf7de7/pkg/apis/core/validation/cloudprofile.go#L34) and [here](https://github.com/gardener/gardener/blob/0921db6fe993a98c2eff652588ded6fedabf7de7/pkg/apis/core/validation/cloudprofile_test.go#L697)